libnopegl/src/backends/gl/pipeline_gl.c

```diff
@@ -604,15 +563,19 @@ int ngli_pipeline_gl_update_attribute(struct pipeline *s, int32_t index, const s
     if (!buffer)
         return 0;
 
-    if (gl->features & NGLI_FEATURE_GL_VERTEX_ARRAY_OBJECT) {
-        const GLuint location = attribute_binding->desc.location;
-        const GLuint size = ngli_format_get_nb_comp(attribute_binding->desc.format);
-        const GLsizei stride = (GLsizei)attribute_binding->desc.stride;
-        const struct buffer_gl *buffer_gl = (const struct buffer_gl *)buffer;
+    const GLuint location = attribute_binding->desc.location;
+    const GLuint size = ngli_format_get_nb_comp(attribute_binding->desc.format);
+    const GLsizei stride = (GLsizei)attribute_binding->desc.stride;
+    const struct buffer_gl *buffer_gl = (const struct buffer_gl *)buffer;
+    if (gl->features & NGLI_FEATURE_GL_VERTEX_ARRAY_OBJECT)
         ngli_glBindVertexArray(gl, s_priv->vao_id);
-        ngli_glBindBuffer(gl, GL_ARRAY_BUFFER, buffer_gl->id);
-        ngli_glVertexAttribPointer(gl, location, size, GL_FLOAT, GL_FALSE, stride, (void*)(uintptr_t)(attribute_binding->desc.offset));
-    }
+    else if (gl->features & NGLI_FEATURE_GL_OES_VERTEX_ARRAY_OBJECT)
+        ngli_glBindVertexArrayOES(gl, s_priv->vao_id);
+    else
+        ngli_assert(0);
```

Comment:
Why not call just `bind_vertex_attribs` instead here? (or remove it)

Reply:
The next PR will rework these functions, adding code to `bind_vertex_attributes()`, removing entirely ngli_pipeline_update_attributes().